syft/pkg/cataloger/kernel/package.go

```diff
@@ -0,0 +1,71 @@
+package kernel
+
+import (
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/source"
+	"strings"
+
+	"github.com/anchore/packageurl-go"
+)
+
+const linuxKernelPackageName = "linux-kernel"
+
+func newLinuxKernelPackage(metadata pkg.LinuxKernelMetadata, locations ...source.Location) pkg.Package {
+	p := pkg.Package{
+		Name:         linuxKernelPackageName,
+		Version:      metadata.Version,
+		Locations:    source.NewLocationSet(locations...),
+		PURL:         packageURL(linuxKernelPackageName, metadata.Version),
+		Type:         pkg.LinuxKernelPkg,
+		MetadataType: pkg.LinuxKernelMetadataType,
+		Metadata:     metadata,
+	}
+
+	p.SetID()
+
+	return p
+}
+
+func newLinuxKernelModulePackage(metadata pkg.LinuxKernelModuleMetadata, locations ...source.Location) pkg.Package {
+	var licenses []string
+	if metadata.License != "" {
+		licenses = []string{metadata.License}
+	} else {
+		licenses = []string{}
+	}
+
+	p := pkg.Package{
+		Name:         metadata.Name,
+		Version:      metadata.Version,
```

Comment:
I found that the version is usually always empty for modules... is there ever a case where this isn't empty? Should we be using the source version instead (which is never empty)?

Reply:
It is. I don't see it all that often, but it is used. If we are to be complete, we need to leave it. The kernelVersion is different.

Reply:
if it is used that's good enough. I know we shouldn't use the kernel version, but the source version looked like another good candidate if the version was always empty.